thrift/thrift_test.go

```diff
@@ -250,8 +257,9 @@ func TestRegisterPostResponseCB(t *testing.T) {
 		}
 
 		called := make(chan struct{})
-		cb := func(method string, response thrift.TStruct) {
+		cb := func(reqCtx context.Context, method string, response thrift.TStruct) {
 			assert.Equal(t, "Call", method)
+			assert.Equal(t, ctx.Value(key), value)
```

Comment:
I don't think this test is testing what you're intending for it to test. So right now, you're using the `contextFn` to create the context that's passed in to the to the test function. So you're basically testing that your test logic creates the context using the function you specified, and that it has the value you set.

What you actually want to test is: if you call `SetContextFn` (https://godoc.org/github.com/uber/tchannel-go/thrift#Server.SetContextFn) to override the context creation to include an extra value, the context you receive inside of the `cb` has that same value.

You probably want your `contextFn` to be used to call `SetContextFn` on a `thrift.Server` right after creating it.